### DISCUSSION TITLE
웹소켓과 socket.io - peterkimzz

### DISCUSSION THREAD

**Discussion by giscus** [peterkimzz/blog#22] [Blog Comments]
# 웹소켓과 socket.io - peterkimzz

예전에 회사 프로젝트를 진행할 때, 지도에 실시간으로 사용자의 위치를 보여주는 기능이 필요해서 socket.io 를 사용해서 구현했던 적이 있습니다.
여태까지 저는 그냥 무지성으로 실시간이면 무조건 socket.io 써야지~ 했었는데, 이번에 코인 시세 모니터링 앱을 

https://www.peterkimzz.com/websocket-vs-socket-io/

**Comment by woorim960** [comment]
멋지십니다..

**Reply by peterkimzz** [reply]
감사합니다!

**Comment by baebyeongil** [comment]
안녕하세요 nodejs 공부중인 24살 한 소년입니다!

공부를 하다 socket.io라는 라이브러리를 사용 할 일이 생겼고 사용한 후
socket.io 에 대해 공부하다 보니

web소켓과 여러 통신 방법들을 알게 되었습니다.

각 통신 방법과 web소켓이 어떤 차이가 있는지 어느정도 알게 되었는데

socket.io의 경우는 실행될 때 해당 브라우저가 web소켓을 지원하는지 확인하는 과정을 거치는 걸로 알고 있는데

이렇게 되면 따로 web소켓을 쓰지 않고 socket.io만 사용하더라도 web소켓을 쓰는 것과 동일한 효과를 얻을 수 있을텐데 굳이 web소켓을 사용하는 이유가 있을 까라는 의문점이 생겼습니다..!

**Reply by peterkimzz** [reply]
안녕하세요. 비유를 들어 답변을 드리자면 Node.js를 사용해도 서버를 구성할 수 있는데요, 실제 프로젝트에서는 순정 그대로 사용하지 않고 대부분 express.js를 사용합니다. 이유는 개발자 편의성을 고려해 이런 저런 기능들이 많이 들어가있고, 조금 더 쉬운 인터페이스로 구성이 되어있기 때문입니다. 

이 편의성은 항상 장점만 있는 것은 아닌데요, 당연하다고 알고 사용하던 것이 실은 아니었고 이 암묵적인 내부 구현으로 에러 원인을 알아내기 어렵게 하기도 합니다. 그리고 당연하지만 용량도 훨씬 크구요.

상황에 맞추어 사용할 수 있다고 생각하시면 될 것 같습니다.

**Reply by baebyeongil** [reply]
좋은 답변감사합니다! 순수 언어만으로도 개발자 실력만 있다면 라이브러리를 쓰지않고도 구현가능한 것과 비슷하게

socket.io를 쓰지 않더라도 ws만으로도 충분히커스텀하여 쓸 수 있지만 사용자 편의성을 생각해서 대부분 socket.io를 쓴다.

대신 그만큼 용량차이도 날거고 라이브러리를 통한 에러 일 수록 직접 수정하거나 잡기 어렵다는 점이 있군요.
